apps/opik-backend/src/main/java/com/comet/opik/infrastructure/bi/InstallationReportService.java

```diff
@@ -0,0 +1,135 @@
+package com.comet.opik.infrastructure.bi;
+
+import com.comet.opik.domain.IdGenerator;
+import com.comet.opik.infrastructure.OpikConfiguration;
+import com.comet.opik.infrastructure.lock.LockService;
+import com.google.inject.ImplementedBy;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import ru.vyarus.dropwizard.guice.module.lifecycle.GuiceyLifecycle;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+
+@ImplementedBy(InstallationReportServiceImpl.class)
+interface InstallationReportService {
+    String NOTIFICATION_EVENT_TYPE = "opik_os_startup_be";
+
+    void reportInstallation();
+}
+
+@Singleton
+@Slf4j
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+class InstallationReportServiceImpl implements InstallationReportService {
+
+    private final LockService lockService;
+    private final IdGenerator generator;
+    private final UsageReportService usageReport;
+    private final OpikConfiguration config;
+    private final Client client;
+
+    public void reportInstallation() {
+
+        String eventType = GuiceyLifecycle.ApplicationStarted.name();
+
+        if (!config.getUsageReport().isEnabled()) {
+            log.info("Usage report is disabled");
+            return;
+        }
+
+        if (StringUtils.isEmpty(config.getUsageReport().getUrl())) {
+            log.warn("Usage report URL is not set");
+            return;
+        }
+
+        var lock = new LockService.Lock("opik-%s".formatted(eventType));
+
+        lockService.executeWithLock(lock, tryToReportStartupEvent(eventType))
+                .subscribeOn(Schedulers.boundedElastic())
+                .onErrorResume(e -> {
+                    log.warn("Didn't reported due to error", e);
+                    return Mono.empty();
+                }).block();
```

Comment:
At the end you block on start-up time, so probably there isn't a great benefit of making all this reactive. It could've been simpler to just go with vanilla Java.

Reply:
I'm adding try catch with block. But the reactive comes from the clock service. You see that the only reactive part is this one.